membership_extension/tests/test_membership.py

```diff
@@ -363,19 +359,17 @@ def test_invoice(self):
                 {
                     "date": fields.Date.today(),
                     "reason": "no reason",
-                    "refund_method": "refund",
+                    "refund_method": "cancel",
                 }
             )
         )
         reversal = move_reversal.reverse_moves()
         refund = self.env["account.move"].browse(reversal["res_id"])
-        refund.journal_id.update_posted = True
-        refund.action_post()  # validate refund
         self.assertEqual("canceled", line.state)
         refund.button_cancel()
         self.assertEqual("paid", line.state)
         refund.button_draft()
-        self.assertEqual("paid", line.state)
+        self.assertEqual("invoiced", line.state)
```

Comment:
Was it wrong before?

Reply:
I am not sure how it was before, but now when the refund is canceled the original invoice is not automatically reconciled with the original payment.

Reply:
I think this is due to payment refactor.